src/history/hs_rec.c

```diff
@@ -911,23 +933,66 @@ __hs_delete_reinsert_from_pos(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, ui
          * the cell. The cell's start timestamp can be cleared during reconciliation if it is
          * globally visible.
          */
-        if (hs_start_ts >= ts || twp->stop_ts >= ts)
+        if (hs_start_ts >= ts || twp->stop_ts >= ts) {
+            if (hs_start_ts == WT_TS_NONE && twp->stop_ts == WT_TS_NONE && ts == WT_TS_NONE)
+                *non_ts_updates = true;
             break;
+        }
     }
-    if (ret == WT_NOTFOUND)
-        return (0);
-    WT_ERR(ret);
+    WT_RET(ret);
 
     /*
      * If we find a key with a timestamp larger than or equal to the specified timestamp then the
      * specified timestamp must be mixed mode.
      */
     WT_ASSERT_ALWAYS(
       session, ts == 1 || ts == WT_TS_NONE, "out-of-order timestamp update detected");
+    return (ret);
+}
+
+/*
+ * __hs_delete_reinsert_from_pos --
+ *     Delete updates in the history store if the start timestamp of the update is larger or equal
+ *     to the specified timestamp from a given cursor position (cursor position should be passed in)
+ *     and optionally reinsert them with ts-1 timestamp.
+ */
+static int
+__hs_delete_reinsert_from_pos(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, uint32_t btree_id,
+  const WT_ITEM *key, wt_timestamp_t ts, bool reinsert, bool no_ts_tombstone,
+  bool error_on_ts_ordering, uint64_t *counter)
+{
+    WT_CURSOR *hs_insert_cursor;
+    WT_CURSOR_BTREE *hs_cbt;
+    WT_DECL_RET;
+    WT_ITEM hs_key, hs_value;
+    WT_TIME_WINDOW hs_insert_tw, *twp;
+    wt_timestamp_t hs_durable_start_ts, hs_durable_stop_ts, hs_start_ts;
+    uint64_t cache_hs_order_lose_durable_timestamp, cache_hs_order_reinsert, cache_hs_order_remove;
+    uint64_t hs_counter, hs_upd_type;
+    uint32_t hs_btree_id;
+#ifdef HAVE_DIAGNOSTIC
+    int cmp;
+#endif
+    char ts_string[5][WT_TS_INT_STRING_SIZE];
+
+    hs_insert_cursor = NULL;
+    hs_cbt = __wt_curhs_get_cbt(hs_cursor);
+    WT_CLEAR(hs_key);
+    WT_CLEAR(hs_value);
+
+    cache_hs_order_lose_durable_timestamp = cache_hs_order_reinsert = cache_hs_order_remove = 0;
+#ifndef HAVE_DIAGNOSTIC
+    WT_UNUSED(key);
+#endif
+
     /*
      * Fail the eviction if we detect any timestamp ordering issue and the error flag is set. We
      * cannot modify the history store to fix the update's timestamps as it may make the history
      * store checkpoint inconsistent.
+     *
```

Comment:
I feel this comment is no longer required. When there non-timestamp updates, the delete call doesn't come into this function.

Reply:
Thanks for spotting, removed here [a1d202d](https://github.com/wiredtiger/wiredtiger/pull/10686/commits/a1d202df26a8439a44350a5330c4fce05b0f89db)